### PR TITLE
+htc #19237 entity#withSizeLimit for Java + add withoutSizeLimit

### LINKED ISSUE #19237: HttpEntity change interface: add withNoLimit
HttpEntity has a method: `HttpEntity.withSizeLimit`, but it has no way to indicate that there should be no limit. It is possible to configure `infinite` in the `application.conf`, but then it would apply to all entities.

A workaround is to set `HttpEntity.withSizeLimit(Long.MaxValue)`, which is near infinite. Having `HttpEntity.withNoLimit()` and/or `HttpEntity.withSizeLimit(-1)` would make it more explicit.

For the discussion see: https://groups.google.com/forum/m/#!topic/akka-user/DIZ2WY97dnA

### PR DESCRIPTION
**Pull request by ktoso**
Resolves #19237

### CODE REVIEW

File: akka-http-core/src/main/java/akka/http/javadsl/model/HttpEntity.java [hunk 1]
```diff
@@ -85,6 +85,45 @@
...
+     * Content-Length and then another limit is applied then this new limit will be evaluated against the new
+     * Content-Length. If the entity is transformed in a way that changes the Content-Length and no new limit is applied
+     * then the previous limit will be applied against the previous Content-Length.
+     *
+     * Note that the size limit applied via this method will only have any effect if the `Source` instance contained
+     * in this entity has been appropriately modified via the `HttpEntity.limitable` method. For all entities created
+     * by the HTTP layer itself this is always the case, but if you create entities yourself and would like them to
+     * properly respect limits defined via this method you need to make sure to apply `HttpEntity.limitable` yourself.
+     */
+    HttpEntity withSizeLimit(long maxBytes);
```
Comment (ktoso) [inline akka-http-core/src/main/java/akka/http/javadsl/model/HttpEntity.java]: moved here to make it accessible to JavaDSL as well.

**Comment by ktoso** [comment]
Please review when you have a second @sirthias @jrudolph

**Comment by johanandren** [comment]
LGTM!

**Comment by ktoso** [comment]
WDYT here @jrudolph ?

**Comment by jrudolph** [comment]
LGTM

**Comment by ktoso** [comment]
Thanks Johannes :)

**Pull request merged**